src/canvastools/ts/CanvasTools/Core/RegionData.ts

```diff
@@ -145,6 +145,39 @@ export class RegionData implements IRegionData, IMovable, IResizable {
         return area;
     }
 
+    public getLineSegments(): Array<[Point2D, Point2D]> {
+        const points = this.regionPoints;
+        if (points.length < 2) {
+            return []
+        }
+        if (points.length === 2) {
+            return [[points[0], points[1]]]
+        }
+        const segments = [];
+        const pointsLength = points.length;
+        const loopLength = pointsLength - 1;
+        for (let i = 0; i < loopLength; i++) {
+            const nextPointIdx = i + 1;
+            if (nextPointIdx < pointsLength) {
+                segments.push([points[i], points[nextPointIdx]])
+            }
+        }
+        if (this.regionType == RegionDataType.Polygon) {
+            // closing line segment from last to first point
+            segments.push([points[pointsLength - 1], points[0]])
+        }
+        return segments;
```

Comment:
If we isolate this code, it seems independent of state (if we just pass in `points`). Wonder if it makes sense to have a geometryutils or lineUtils class for pure mathematical calcs? Then we can pass `return lineSegments(this.regionPoints, this.regionType)`

Reply:
Yea, I think there are a lot of opportunities to extract common code into utilities. I've been hewing more closely to the existing code architecture which has quite a few very similar class/code functions with slight tweaks between them.
I'm hoping to have some time at the end to refactor to DRY it up.

Reply:
Cool sounds good. Not a blocking comment